test/linalg/matmul.jl

```diff
@@ -164,6 +165,21 @@ let A = [1+2im 3+4im; 5+6im 7+8im], B = [2+7im 4+1im; 3+8im 6+5im]
     end
 end
 
+#vecdot for generic reals as well
+let A = [1 3; 5 7], B = [2 4; 3 6]
+    @test vecdot(A,B) == dot(vec(A),vec(B)) == vecdot_(A,B) == vecdot(float(A),float(B))
+    @test vecdot(Int[], Int[]) == 0 == vecdot_(Int[], Int[])
+    @test_throws MethodError vecdot(Any[], Any[])
```

Comment:
I'm being dense, but that line seems to require a `BlasReal` and this is `Any`. The test seems to call https://codecov.io/github/JuliaLang/julia/base/linalg/generic.jl?ref=1eacbbbb9a686292a12e89bcc4d9559c22928b83#l-270, which is (mostly) covered, and which ends up being a test of whether `zero(Any)` is defined.